### PR TITLE
Small refactorings to allow for RenderTarget support for WebGPU

### PR DESCRIPTION
**Pull request by mvaligursky**
- multiple small changes

### CODE REVIEW

File: src/resources/parser/texture/img.js [hunk 1]
```diff
@@ -23,7 +26,9 @@ class ImgParser {
         // - Firefox doesn't support options parameter to createImageBitmap (see https://bugzilla.mozilla.org/show_bug.cgi?id=1533680)
         // - Safari supports ImageBitmap only as experimental feature.
         // this.useImageBitmap = typeof ImageBitmap !== 'undefined' && /Firefox/.test(navigator.userAgent) === false;
-        this.useImageBitmap = false;
+        // WebGPUTexture expects ImageBitmap.
+        const isWebGPU = registry?._loader?._app?.graphicsDevice?.deviceType === DEVICETYPE_WEBGPU;
```
Comment (slimbuck) [inline src/resources/parser/texture/img.js:30]: might be better to pass device type to handlers?

File: src/graphics/render-target.js [hunk 2]
```diff
@@ -117,7 +117,8 @@ class RenderTarget {
         }
 
         // device, from one of the buffers
-        this._device = this._colorBuffer?.device || this._depthBuffer?.device || options.graphicsDevice;
+        const device = this._colorBuffer?.device || this._depthBuffer?.device || options.graphicsDevice;
+        this._device = device;
         Debug.assert(this._device, "Failed to obtain the device, colorBuffer nor depthBuffer store it.");
```
[the remark this answers is not in the document]
Comment (mvaligursky) [reply src/graphics/render-target.js]: done

File: src/resources/parser/texture/img.js (hunk 1 above)
Comment (mvaligursky) [reply src/resources/parser/texture/img.js:30]: this is temporary I think, as you wanted to look at enabling this for webgl too?

File: src/graphics/render-target.js [hunk 3]
```diff
@@ -117,7 +117,8 @@ class RenderTarget {
...
 
         this._samples = (options.samples !== undefined) ? Math.min(options.samples, this._device.maxSamples) : 1;
```
[the remark this answers is not in the document]
Comment (mvaligursky) [reply src/graphics/render-target.js:124]: done, thanks

File: src/graphics/graphics-device.js [hunk 4]
```diff
@@ -270,6 +271,32 @@ class GraphicsDevice extends EventHandler {
...
 
+    /**
+     * Initialize render target before it can be used.
+     *
+     * @param {RenderTarget} target - The render target to be initialized.
+     * @ignore
+     */
+    initRenderTarget(target) {
+
+        if (target.impl.initialized) return;
```
[the remark this answers is not in the document]
Comment (mvaligursky) [reply src/graphics/graphics-device.js]: done, thanks

**Review by slimbuck — APPROVED** [review]

**Pull request merged**